### PR TITLE
section over pr. build correctly.

### PR DESCRIPTION
**Pull request by zainen**
part of the second lint pr to sift through non-breaking changes 

this portion is smaller as well as builds correctly 

- [ ] Your code builds cleanly without any errors or warnings
- [ ] You have added unit tests
- [ ] You have added integration tests (if relevant/appropriate)
- [ ] All public methods or types have TypeDoc coverage with a complete description, and ideally an @example
- [ ] You have added or updated corresponding documentation
- [ ] If relevant, you have written a first draft summary describing the change for inclusion in Release Notes. 

## Release Note Draft Snippet

__If relevant, please write a summary of your change that will be suitable for
inclusion in the Release Notes for the next Taquito release.__

### CODE REVIEW

File: packages/taquito-michelson-encoder/src/schema/storage.ts [hunk 1]
```diff
@@ -34,22 +34,23 @@ function collapse(val: Token['val'] | any[], prim: string = PairToken.prim): Tok
...
         },
       ],
     };
   }
-  return val;
+  return extended;
 }
 
-function deepEqual(a: Token['val'] | any[], b: Token['val'] | any[]): boolean {
+function deepEqual(a: MichelsonV1Expression | MichelsonV1Expression[], b: MichelsonV1Expression | MichelsonV1Expression[]): boolean {
```
Comment (roxaneletourneau) [inline packages/taquito-michelson-encoder/src/schema/storage.ts:53]: I am wondering if `| MichelsonV1Expression[]` is necessary as it is already part of `MichelsonV1Expression`

File: packages/taquito-michelson-encoder/src/schema/storage.ts [hunk 2]
```diff
@@ -236,27 +249,29 @@ export class Schema {
    * @param valueType type of value to look for
    *
    */
-  FindFirstInTopLevelPair<T extends MichelsonV1Expression>(storage: any, valueType: any) {
```
Comment (roxaneletourneau) [inline packages/taquito-michelson-encoder/src/schema/storage.ts:239]: Removing `<T extends MichelsonV1Expression>` seems to be a breaking change for the users. I am not sure we should make that change

File: packages/taquito-michelson-encoder/src/tokens/pair.ts [hunk 3]
```diff
@@ -55,7 +56,7 @@ export class PairToken extends ComparableToken {
   static prim: 'pair' = 'pair';
 
   constructor(
-    val: { prim: string; args: any[]; annots: any[] } | any[],
+    val: MichelsonV1ExpressionExtended | MichelsonV1Expression[],
```
Comment (roxaneletourneau) [inline packages/taquito-michelson-encoder/src/tokens/pair.ts:59]: consider using `MichelsonV1Expression`

File: packages/taquito/src/contract/rpc-contract-provider.ts [hunk 4]
```diff
@@ -124,12 +124,12 @@ export class RpcContractProvider
    *
    * @see https://tezos.gitlab.io/api/rpc.html#get-block-id-context-big-maps-big-map-id-script-expr
    */
-  async getBigMapKeyByID<T>(
+  async getBigMapKeyByID(
```
Comment (roxaneletourneau) [inline packages/taquito/src/contract/rpc-contract-provider.ts:127]: removing <T> seems to be a breaking change for the users. I don't think we should make that change.

File: packages/taquito-michelson-encoder/src/schema/storage.ts (hunk 2 above)
Comment (zainen) [reply packages/taquito-michelson-encoder/src/schema/storage.ts:239]: 👍  ok I'll keep away form changes like these!

File: packages/taquito-tzip12/src/tzip12-contract-abstraction.ts [hunk 5]
```diff
@@ -179,7 +179,7 @@ export class Tzip12ContractAbstraction {
     const bigmapTokenMetadataId = await this.findTokenMetadataBigMap();
     let pairNatMap;
     try {
-      pairNatMap = await this.context.contract.getBigMapKeyByID<any>(
+      pairNatMap = await this.context.contract.getBigMapKeyByID<{token_info: BigMapId}>(
```
Comment (roxaneletourneau) [inline packages/taquito-tzip12/src/tzip12-contract-abstraction.ts:182]: Not sure the type is right, should be something like this: {token_info: MichelsonMap<string, string>}

File: packages/taquito-tzip16/src/handlers/tezos-storage-handler.ts [hunk 6]
```diff
@@ -33,17 +32,17 @@ export class TezosStorageHandler implements Handler {
       parsedTezosStorageUri.contractAddress || contractAbstraction.address,
       'head'
     );
-    const bigMapId = Schema.fromRPCResponse({ script }).FindFirstInTopLevelPair<BigMapId>(
+    const bigMapId = Schema.fromRPCResponse({ script }).FindFirstInTopLevelPair<MichelsonV1ExpressionBase>(
```
Comment (roxaneletourneau) [inline packages/taquito-tzip16/src/handlers/tezos-storage-handler.ts:35]: Type should be BigMapId

File: packages/taquito-tzip16/src/handlers/tezos-storage-handler.ts [hunk 7]
```diff
@@ -33,17 +32,17 @@ export class TezosStorageHandler implements Handler {
...
       script.storage,
       typeOfValueToFind
     );
 
-    if (!bigMapId) {
+    if (!bigMapId || !bigMapId.int) {
       throw new BigMapMetadataNotFound();
     }
-
+    // TODO check this
```
Comment (roxaneletourneau) [inline packages/taquito-tzip16/src/handlers/tezos-storage-handler.ts:43]: Consider removing // TODO check this
or adding more detail on what needs to be checked

File: packages/taquito-tzip16/src/tzip16-contract-abstraction.ts [hunk 8]
```diff
@@ -43,12 +42,12 @@ export class Tzip16ContractAbstraction {
   }
 
   private async findMetadataBigMap(): Promise<BigMapAbstraction> {
-    const metadataBigMapId = this.constractAbstraction.schema.FindFirstInTopLevelPair<BigMapId>(
```
Comment (roxaneletourneau) [inline packages/taquito-tzip16/src/tzip16-contract-abstraction.ts:46]: Type should be BigMapId

**Review by roxaneletourneau — APPROVED** [review]
Looks good to me!

**Pull request merged**